pkg/certificate/providers/vault/certificate_manager_test.go

```diff
@@ -147,11 +155,27 @@ func TestIssueCertificate(t *testing.T) {
 	for _, tc := range testCases {
 		t.Run(tc.description, func(t *testing.T) {
 			tassert := assert.New(t)
-			_, err = cm.IssueCertificate(certificate.NewCertOptionsWithFullName(tc.cn, tc.vP))
+			o := certificate.NewCertOptionsWithTrustDomain(tc.cn, "cluster.local", tc.vP, tc.spiffeEnabled)
+			cert, err := cm.IssueCertificate(o)
+
 			if tc.wantErr {
 				tassert.Error(err, "expected error, got nil")
-			} else {
-				tassert.NoError(err, "did not expect error, got %v", err)
+				return
+			}
+
+			tassert.NoError(err, "did not expect error, got %v", err)
+			xCert, err := certificate.DecodePEMCertificate(cert.CertChain)
+			tassert.NoError(err, "did not expect error, got %v", err)
+			tassert.Equal(tc.cn+".cluster.local", xCert.Subject.CommonName)
+
+			if tc.spiffeEnabled {
+				uri := xCert.URIs[0]
+				tassert.Equal("spiffe://cluster.local/"+tc.cn, uri.String())
+
+				// Vaults test mock doesn't create private keys properly?
+				// svid, err := x509svid.Parse(cert.CertChain, cert.PrivateKey)
+				// tassert.NoError(err)
+				// tassert.Equal("spiffe://cluster.local/"+tc.cn, svid.ID.String())
```

Comment:
Not sure if this is meant to be included here

Reply:
yes sorry, missed cleaning this up.   We have coverage in other places so will remove.  Interesting the certs produced here aren't valid.

Reply:
It's a good callout. Do you think it's worth tracking in a bug?

Reply:
It's more a bug or issue with Vault's testing api. I don't know that this is a critical check and don't think it is probably worth tracking down the issue in vault.  This was more of a sanity on my part to make sure the certs we produce pass SPIFFE validation.  We have that check in Tressor so I feel like we have it covered if anyone was to change the URI SAN implementation it would be caught.